app/routes.py

```
@@ -131,7 +131,22 @@ def is_hash_valid(timed_hash: str) -> bool:
+    """Return True if no role restrictions or the user's role is allowed."""
+
+    if not roles:
+        return True
+    if isinstance(user, dict):
+        user_role = user.get("role")
+    else:
+        user_role = getattr(user, "role", None)
+    return user_role in roles
+
+
+def login_required(f=None, *, roles=None):
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)